src/GafferCycles/IECoreCyclesPreview/Renderer.cpp

```diff
@@ -2413,6 +2405,49 @@ ccl::DeviceInfo firstCPUDevice()
 	return ccl::DeviceInfo();
 }
 
+ccl::DeviceInfo matchingDevices( const std::string &pattern, int threads, bool background )
+{
+	ccl::vector<ccl::DeviceInfo> devices;
+	std::unordered_map<ccl::DeviceType, int> typeIndices;
+	for( const auto &device : ccl::Device::available_devices() )
+	{
+		const string typeString = ccl::Device::string_from_type( device.type );
+		const int typeIndex = typeIndices[device.type]++;
+		const string name = fmt::format( "{}:{:02}", typeString, typeIndex );
```

Comment:
I was thinking if Gaffer is working with any [fancy ML frameworks](https://pytorch.org/docs/stable/tensor_attributes.html#torch.device), it looks like they specify device with `{}:{:01}` (or whatever width of 1 is in fmt strings) so I am wondering if that should be supported as well for consistency if someone runs some python scripts, just a thought (I think padding is better though).

Reply:
At the moment I'm just trying to keep the behaviour the same while simplifying the code as much as I can, in preparation for it getting more complicated again when this option buffering gets introduced. Happy to tweak device formatting in the future if there's a good case for it though...